trio/_subprocess_platform/__init__.py

```diff
@@ -4,6 +4,7 @@
 import sys
 from typing import Optional, Tuple, TYPE_CHECKING
 
+import trio
```

Comment:
I'm surprised this doesn't lead to a circular import problem. Also this kind of change seems out of place in this PR?

Reply:
It looks to me like it's potentially resolving a circular import, since previously we imported `lowlevel` directly (further below).

Also in general there has been a semi-recent trend towards having Trio use its own APIs under their public names where possible, because that makes it easier for others to understand the code.

Reply:
It's not a circular import because it just pulls in the `trio` module object – that module object isn't fully populated yet, so trying to access attributes on it is likely to fail. But we're not touching the module's attributes during the import, just the module object itself. We don't access attributes until later, when functions in this module are actually called, by which point the imports have all finished and the `trio` object is all filled in.